tests/test_output.py

```diff
@@ -89,8 +88,7 @@ def tearDown():
                 os.unlink(join(config_dir, cf))
 
         for output_file in glob(join(output_dir, 'test_{}_*'.format(suffix))) \
-            + glob(join(output_dir, 'test_{}.log'.format(suffix))) \
-                + glob(join(output_dir, 'test_majority_class_custom_learner_*')):
+                           + glob(join(output_dir, 'test_{}.log'.format(suffix))):
```

Comment:
It looks like this isn't a `glob` pattern? I think it could be dealt with elsewhere. I'm not sure if this has anything to do with the build failure, but it's probably better to remove it above (or below) if it exists.

Reply:
I don't think this was the issue. But line 90 (`'test_{}_*'.format(suffix)))`) was also matching the log filename of my test, despite me trying to be super specific about the filename I used. I've changed it now. 

Btw, is the `tearDown` method called after each test? Because, that would explain the race condition.

Reply:
Yes, I believe that's right.